presto-parquet/src/main/java/com/facebook/presto/parquet/writer/ParquetWriter.java

```diff
@@ -98,11 +100,15 @@ public ParquetWriter(OutputStream outputStream,
 
         this.messageType = requireNonNull(messageType, "messageType is null");
 
-        ParquetProperties parquetProperties = ParquetProperties.builder()
+        Builder parquetPropertiesBuilder = ParquetProperties.builder()
                 .withWriterVersion(writerOption.getWriterVersion())
                 .withPageSize(writerOption.getMaxPageSize())
-                .withDictionaryPageSize(writerOption.getMaxDictionaryPageSize())
-                .build();
+                .withDictionaryPageSize(writerOption.getMaxDictionaryPageSize());
+        if (!DEFAULT_MAX_PAGE_SIZE.equals(DataSize.succinctBytes(writerOption.getMaxPageSize())) ||
```

Comment:
Can you please add a comment explaining why we must do this?

Reply:
Sure, done. Please take a look when convenient, thanks.